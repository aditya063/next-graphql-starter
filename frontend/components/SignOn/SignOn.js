import WithUser from '../wrappers/WithUser/WithUser';
import SignOut from './SignOut/SignOut';
import PopUps from './PopUps/PopUps';
import * as Styled from './SignOn.style';

class SignOn extends React.PureComponent {
  state = {
    popUp: ''
  };

  render() {
    const { popUp } = this.state;

    return (
      <WithUser>
        {me => (
          <Styled.divSignOn>
            {me ? (
              <SignOut />
            ) : (
              <>
                <Styled.purpleLinkBtn
                  onClick={() => this.setState({ popUp: 'signIn' })}
                >
                  Sign In
                </Styled.purpleLinkBtn>

                <Styled.orangeLinkBtn
                  onClick={() => this.setState({ popUp: 'signUp' })}
                >
                  Sign Up
                </Styled.orangeLinkBtn>

                {popUp && (
                  <PopUps
                    popUp={popUp}
                    setPopUp={popUpName => this.setState({ popUp: popUpName })}
                  />
                )}
              </>
            )}
          </Styled.divSignOn>
        )}
      </WithUser>
    );
  }
}

export default SignOn;
