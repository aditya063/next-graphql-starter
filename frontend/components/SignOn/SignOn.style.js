import styled from 'styled-components';
import { purpleBtn, orangeBtn } from '../../styles/button.style';

export { purpleBtn, orangeBtn };

export const div = styled.div``;

export const divPopup = styled.div`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  background: rgba(0, 0, 0, 0.5);
`;

export const divOuter = styled.div`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
`;

export const divInner = styled.div`
  position: absolute;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 20px;
  background-color: white;
`;
