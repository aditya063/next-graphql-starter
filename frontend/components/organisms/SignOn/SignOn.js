import SignIn from '../SignIn/SignIn';
import SignUp from '../SignUp/SignUp';
import { title } from '../../../constants';
import * as sc from './SignOn.style';

const SignOn = () => (
  <sc.SignOn>
    <sc.Header>
      <sc.Img src='images/ideabox.png' alt='ideabox' />

      <sc.Title>{title}</sc.Title>
    </sc.Header>

    <sc.Content>
      <SignIn />

      <sc.HR />

      <SignUp />
    </sc.Content>
  </sc.SignOn>
);

export default SignOn;
