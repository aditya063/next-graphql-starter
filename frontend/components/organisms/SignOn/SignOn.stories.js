// import { action } from '@storybook/addon-actions';
import allCombos from '../../../.storybook/allCombos';
import SignOn from './SignOn';

export default { title: 'Components|Organisms', component: SignOn };

const data = {
  // id: ['1']
};

// const actions = {
//   onClick: action('onClick')
// };

export const signOn = () => allCombos(SignOn, data);
