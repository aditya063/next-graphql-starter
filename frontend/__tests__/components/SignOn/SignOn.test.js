import React from 'react';
import { shallow } from 'enzyme';
import SignOn from '../../../components/SignOn/SignOn';

describe('SignOn', () => {
  let mockProps;
  let wrapper;

  beforeEach(() => {
    jest.resetAllMocks();
    mockProps = {};
    wrapper = shallow(<SignOn {...mockProps} />, {
      disableLifecycleMethods: true
    });
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
