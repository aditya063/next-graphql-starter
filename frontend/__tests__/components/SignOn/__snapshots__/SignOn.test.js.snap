// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignOn matches snapshot 1`] = `
<Component>
  <Component />
</Component>
`;
